### DISCUSSION TITLE
How to access a value of a Variable by it´s browsepath?

### DISCUSSION THREAD

**Discussion by maki1-1** [FreeOpcUa/python-opcua#1338] [Q&A] [answered]
Hello, i´m trying to access a value of a variable via its browsepath.

So far, I did it via its nodeid. However, due to some circumstances, I would rather get the value via the browsepath.

Thus, I created from serverside

`BALANCE_A1= Variables.add_object('ns=2;g=d83e2d46-4eab-463b-ba3b-bc2b8ddfbca6', "BALANCE_A1")
Value_BALANCE_A1=BALANCE_A1.add_variable('ns=2;g=a60830a6-3651-44f2-b5af-313e14da54df',"Value_BALANCE_A1",2)`

and the paths of those are 
`"0:Objects,0:Units,0:FM_1,0:Variables,0:BALANCE_A1" and "0:Objects,0:Units,0:FM_1,0:Variables,0:BALANCE_A1,0:Value_BALANCE_A1"`

Thus, I tried to access it via 

`myvar=client.get_node(["0:root,0:Objects,0:Units,0:FM_1,0:Variables,0:BALANCE_A1"])`
which throws the error 
`argument to node must be a NodeId object or a string defining a nodeid found ['0:root,0:Objects,0:Units,0:FM_1,0:Variables,0:BALANCE_A1'] of type <class 'list'>`

Thus, I guess acccessing it via the browsepath is not possible on that way.

So i tried 

`myvar=root.get_child(["0:Objects,0:Units,0:FM_1,0:Variables,0:BALANCE_A1,0:Value_BALANCE_1"])`
which throws
`"The requested operation has no match to return."(BadNoMatch)`

So what is the best way to read a value from a variable by using its browsepath?

Any help is welcome!

**Comment by AndreasHeine** [comment, marked as answer]
https://reference.opcfoundation.org/v104/Core/docs/Part4/5.8.4/

https://github.com/FreeOpcUa/python-opcua/blob/3012acf9047aad3acc2674e9008f0d1eee680c25/opcua/client/ua_client.py#L423

**Reply by maki1-1** [reply]
Thanks for your reply. Would it be too much to ask if you can give me an example? I tried 

server.translate
and
client.translate,

but both are not recognized

**Reply by AndreasHeine** [reply]
"client.ua_client."
or
"server.iserver.aspace."

hint:
debugger + breakpoint

**Reply by oroulet** [reply]
There are example in almost all files in examples directory.. your last example uses a long string, you need a list of strings and it will work

**Reply by maki1-1** [reply]
That was the mistake, I´m blind. Thank you a lot
